src/utils.c

```
@@ -196,10 +303,77 @@ int parse_cuda_visible_env() {
+        cuda_to_nvml_map_count = visible_count;
+    }
     return 0;
 }
 
+static nvmlReturn_t map_cuda_devices_to_nvml_by_pci(void) {
+    unsigned int mapped_devices[CUDA_DEVICE_MAX_COUNT];
+    int cuda_device_count = 0;
+    int ordinal;
+    CUresult cuda_result;
+
+    cuda_result = CUDA_OVERRIDE_CALL(cuda_library_entry, cuDeviceGetCount,
```

Cppcheck (2.21.0)

[warning] If resource allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfResources)